### ISSUE TITLE
ability to support search

### ISSUE THREAD

**Issue by chris-annino-sp** [microsoftgraph/msgraph-sdk-go-core#35]
I am using this code
```
	searchStr := fmt.Sprintf("appId:%s", applicationId)
	client := msgraphsdk.NewGraphServiceClient(adapter)
	spResponse, err := client.ServicePrincipals().Get(&serviceprincipals.ServicePrincipalsRequestBuilderGetOptions{
		Q: &serviceprincipals.ServicePrincipalsRequestBuilderGetQueryParameters{
			Search: &searchStr,
		},
	})
```
to attempt to generate a url like `/v1.0/servicePrincipals?$search="appId:applicationId"`
however inside microsoft.kiota.abstractions/request_information.go it appears to use reflection to determine the fieldName. The end result of the RawQuery in the request object is `search=appId%3Aca1ce348-e18e-4aa0-93fd-2901f9fc6ee8` w/o the dollar.

is there a workaround for this? am I using the API incorrectly?

I realize this is probably more about kiota and less about this SDK but would appreciate some comments or direction on this issue. Thanks!

**Comment by baywet** [comment]
Hi @chris-annino-sp 
Thanks for reaching out and thanks for trying the Go SDK.
Could you double check your dependencies are up to date by running `go get -u github.com/microsoftgraph/msgraph-sdk-go-core` please?
There's a middleware handler in charge of adding the missing dollar signs to query parameters.
https://github.com/microsoftgraph/msgraph-sdk-go-core/blob/ec4b64e7eb2577ec0467aff3cfc6d85b0b043049/graph_odata_query_handler.go#L50

**Comment by chris-annino-sp** [comment]
@baywet thanks for the tips.
Took me a while, as I had to upgrade my env to go 1.17 and resolve some module conflicts. Then got hit by the lack of error handling in the sdk, I know that's being worked on. In the end, I need to add the special header :) It works!
```
	spResponse, err := client.ServicePrincipals().Get(&serviceprincipals.ServicePrincipalsRequestBuilderGetOptions{
		H: map[string]string{"ConsistencyLevel": "eventual"},
		Q: &serviceprincipals.ServicePrincipalsRequestBuilderGetQueryParameters{
			Search: &searchStr,
		},
	})
```

**Comment by baywet** [comment]
Great news!
On the errors front, we've made some progress.
First, we've decided on a design https://github.com/microsoft/kiota/issues/411
Second, I've put a pull request together to update the conversion library https://github.com/microsoft/OpenAPI.NET.OData/pull/167

The next step will be to implement the design changes in the generator. And core librairies offered by kiota. Don't hesitate to have a look at all of that and provide feedback.

Since your problem is resolved, I'm going to close this issue.
